### DISCUSSION TITLE
why `type Maybe<T> = T | null` when it could be `type Maybe<T> = T | undefined` for TS?

### DISCUSSION THREAD

**Discussion by Morriz** [dotansimha/graphql-code-generator#7895] [Q&A]
I see issues between two code generators I use, and yours uses `null` for typescript instead of `undefined`. Usually `null` is frowned upon when `undefined` is meant, and `null` indicates a special case of being nullable. However, after observing the output I conclude that nullable is not the intent.

So wouldn't it be on spec to change
```
type Maybe<T> = T | null
```
to
```
type Maybe<T> = T | undefined
```
?
(I am using `"@graphql-codegen/cli": "2.6.2"` btw.)

I am probably not aware of the reasoning behind this, and would love to be pointed there.

**Comment by NickSchmitt** [comment]
I have the same question. Is `null` the default empty value in GraphQL?

**Reply by ardatan** [reply]
Yes GraphQL returns `null` for empty values not `undefined`.
